WordPress/WordPressTest/Domains/AllDomainsListItemViewModelTests.swift

```diff
@@ -0,0 +1,121 @@
+import XCTest
+
+@testable import WordPress
+
+final class AllDomainsListItemViewModelTests: XCTestCase {
+
+    func testMappingWithDefaultInput() throws {
+        self.testMapping(
+            domain: try .make(),
+            expectedViewModel: .make()
+        )
+    }
+
+    func testMappingWithDomainOnlySite() throws {
+        self.testMapping(
+            domain: try .make(isDomainOnlySite: true),
+            expectedViewModel: .make(description: nil)
+        )
+    }
+
+    func testMappingWithEmptyBlogNameDomain() throws {
+        self.testMapping(
+            domain: try .make(blogName: ""),
+            expectedViewModel: .make(description: Domain.Defaults.siteSlug)
+        )
+    }
+
+    func testMappingWithUnregisteredDomain() throws {
+        self.testMapping(
+            domain: try .make(hasRegistration: false),
+            expectedViewModel: .make(expiryDate: nil)
+        )
+    }
+
+    func testMappingWithValidDomain() throws {
+        let futureDate = Date.init(timeIntervalSinceNow: 365 * 24 * 60 * 60)
+        let iso8601Date = ViewModel.DateFormatters.iso8601.string(from: futureDate)
+        let humanReadableDate = ViewModel.DateFormatters.humanReadable.string(from: futureDate)
+        self.testMapping(
+            domain: try .make(expiryDate: iso8601Date),
+            expectedViewModel: .make(expiryDate: "Renews \(humanReadableDate)")
+        )
+    }
+
+    func testMappingWithSiteRedirectDomain() throws {
+        let expectedURL = URL(string: "\(ViewModel.domainManagementBasePath)/example1.com/redirect/exampleblog1.wordpress.com")
+        self.testMapping(
+            domain: try .make(type: "redirect"),
+            expectedViewModel: .make(wpcomDetailsURL: expectedURL)
+        )
+    }
+
+    func testMappingWithTransferDomain() throws {
+        let expectedURL = URL(string: "\(ViewModel.domainManagementBasePath)/example1.com/transfer/in/exampleblog1.wordpress.com")
+        self.testMapping(
+            domain: try .make(type: "transfer"),
+            expectedViewModel: .make(wpcomDetailsURL: expectedURL)
+        )
+    }
+
+    private func testMapping(domain: Domain, expectedViewModel: ViewModel) {
+        XCTAssertEqual(ViewModel(domain: domain), expectedViewModel)
+    }
```

Comment:
Nitpick. In XCTest, functions that start with "test" are picked up by the test runner. I don't think this will because it has arguments, too, but IMHO it would be clearer if it had a different name. For example given that it wraps an assertion:

```suggestion
    private func assert(domain: Domain, mapsTo expectedViewModel: ViewModel) {
        XCTAssertEqual(ViewModel(domain: domain), expectedViewModel)
    }
```

Reply:
Resolved in bce644663ff0e8bd1fa063fc32c48ead9854794a